### DISCUSSION TITLE
How to correctly transform path within an ActiveSelection (being moved)

### DISCUSSION THREAD

**Discussion by dmuy** [fabricjs/fabric.js#10341] [Q&A]
Hello,

I've been working on a project using fabricjs as the "drawing tool" backbone. So the issue I'm facing, is I don't know how to update the path points correctly when it is being moved within an ActiveSelection object (as shown in the video below).

https://github.com/user-attachments/assets/7d6d862d-a3f7-4cc4-9e75-a205276033d6

The code I have below works if I move the line individually (above video is not calling `_movePath` on selection move, just the default behavior hence why the labels are not being moved).
```typescript
_movePath(transformMatrix?: Array<any>) {
    const matrix = transformMatrix ?? this.calcTransformMatrix(true)
    const { path, pathOffset } = this
    const _path = fabric.util.transformPath(path, matrix, pathOffset)

    this.path[0][1] = _path[0][1]
    this.path[0][2] = _path[0][2]
    this.path[1][1] = _path[1][1]
    this.path[1][2] = _path[1][2]
    this.path[1][3] = _path[1][3]
    this.path[1][4] = _path[1][4]

    this._updateLineCoords()
}
...
_updateLineCoords() {
    const dims = this._calcDimensions()
    this.setOptions({
        width: dims.width,
        height: dims.height,
        left: dims.left,
        top: dims.top,
        pathOffset: {
            x: dims.width / 2 + dims.left,
            y: dims.height / 2 + dims.top
        },
        dirty: true
    })
    this.setCoords()
}
```
But if I call `_movePath` when the active selection is moving, it positions the lines and labels weirdly.

https://github.com/user-attachments/assets/99d67b16-c74e-41e5-9118-0544b42fe79a

For context, the line shown in the video above is created using a custom class extending `fabric.Path` and I'm using version 5 (still). I've already searched but haven't found any similar issue with mine - just how to update path points (when moving individually). Any help would be much appreciated 🙏

**Comment by asturur** [comment]
The path inside the active selection is inherithing the transform of the active selection ( including translation ) at render time.
If you apply that transform to the path points while the path is still inside the active selection, the active selection translation will happen twice.

Is unclear to me what is not working actually and what do you expect to do.

I wouldn't touch at all a path when is moving or moving inside an active selection. What is the reason to transform it every single movement?

What effect are you trying to achieve?

**Reply by dmuy** [reply]
If I don't call `_movePath` when it is moved within an active selection, the (custom) controls and label (fabric.Text) that I placed doesn't get updated (as shown on the first video capture at 0:20) since I use the path's points to position them. 🤔 Is there another way perhaps to position them correctly?

**Reply by asturur** [reply]
No path points are good, but paths point are good also without every changing them, you have just to transform the back to current canvas coordinates.
Did you check how the for the path controls point works for positioning? those controls follow the path points when you move it around.

https://fabricjs.com/api/namespaces/controlsutils/functions/createpathcontrols/

follow the link to the code and the look for pathPositionHandler

**Reply by dmuy** [reply]
Thank you so much for the help @asturur 🙏 appreciate it. The positioning logic on `pathPositionHandler` helped with my issue - no need for me to update the path points every move now.

**Reply by dmuy** [reply]
Hmm it resolved my issue on moving the path labels and controls but when I move a control it resets the path points.

https://github.com/user-attachments/assets/080393aa-a652-4fc2-8099-71a9f1dee02a

**Reply by asturur** [reply]
Sorry with videos we are going nowhere.
Either you can reproduce this path in a fiddle/codepen or i can't help

**Reply by dmuy** [reply]
Hello, this is the minimal reproduction of the custom line class I made: [jsfiddle](https://jsfiddle.net/u9tb05s1/1/)
Let me know your thoughts.
